tensorflow/core/kernels/random_op_gpu.cu.cc

```diff
@@ -141,7 +141,7 @@ struct FillPhiloxRandomKernel<Distribution, false> {
       const typename Distribution::ResultType samples = dist(&gen);
       copier(&data[offset], samples);
 
-      offset += (total_thread_count - 1) * kGroupSize;
+      offset += total_thread_count * kGroupSize;
       gen.Skip(total_thread_count - 1);
```

Comment:
Can you examine the state of the kernel for those two threads using cuda-gdb or printfs?  I'd like to make sure we understand what is actually going on - there's probably something we're missing.